modules/admin_manual/pages/configuration/files/file_versioning.adoc

```diff
@@ -1,37 +1,135 @@
+This is done by automatic background jobs which clean up the versions following
+a specific pattern. This pattern defines the expiration date for each backup version.
+
+=== Default Versions Delete Patterns
 
-The Versions app (files_versions) expires old file versions
-automatically to ensure that users don’t exceed their storage quotas.
 This is the default pattern used to delete old versions:
 
-* For the first second we keep one version
-* For the first 10 seconds ownCloud keeps one version every 2 seconds
-* For the first minute ownCloud keeps one version every 10 seconds
-* For the first hour ownCloud keeps one version every minute
-* For the first 24 hours ownCloud keeps one version every hour
-* For the first 30 days ownCloud keeps one version every day
-* After the first 30 days ownCloud keeps one version every week
+* For the last second we keep one version
+* For the last 10 seconds ownCloud keeps one version every 2 seconds
+* For the last minute ownCloud keeps one version every 10 seconds
+* For the last hour ownCloud keeps one version every minute
+* For the last 24 hours ownCloud keeps one version every hour
+* For the last 30 days ownCloud keeps one version every day
+* If the versions are older than 30 days ownCloud keeps one version every week
 
 The versions are adjusted along this pattern every time a new version is
-created.
+created and the background job was executed.
 
-The Versions app never uses more that 50% of the user’s storage quota.
+=== Example
+
+[cols="2,3", options="header"]
+|===
+|Time Period before last Expiration
+|Maximum Number of Versions:
+
+|1 second
+|1
+
+|10 seconds
+|5
+
+| 1 minute
+| 6
+
+| 1 hour
+| 59
+
+| 1 day
+| 23
+
+| 30 days
+| 30
+|===
+
+WARNING: The versions app never uses more that 50% of the user’s storage quota.
 If the stored versions exceed this limit, ownCloud deletes the oldest
 file versions until it meets the disk space limit again.
 
+TIP: Adjust the `'versions_retention_obligation'` setting in `config.php` to avoid
+filling up the user's quota.
+
+== Change the Expiration Settings
+
 You may alter the default pattern in `config.php`. The default setting
 is `auto`, which sets the default pattern:
 
 ....
 'versions_retention_obligation' => 'auto',
 ....
 
-Additional options are:
+==== Possible config values are:
+[cols="1a,3"]
+|===
+|* `D, auto`
+|Keep versions at least for D days, apply expiration rules to all versions that are older than D days
+
+|* `auto, D`
+|Delete all versions that are older than D days automatically, delete other versions according to expiration rules
+
+|* `D1, D2`
+|Keep versions for at least `D1` days and delete when they exceed `D2` days.
+
+|* `disabled`
+|Disable Versions; no files will be deleted.
+|===
+
+=== Example 1:
+
+Keep all versions for at least 10 days, apply expiration rules to all versions that are older than 10 days.
+This will keep a lot more versions during the last 10 days compared to the default pattern.
+
+....
```

Comment:
```
[console,php]
----
command...
----
```
When move over that block, it tells you it is php
pls also for the other examples

Reply:
How do you see it? I tried in the local antora build with Firefox, don't see any hint that it is php.

Reply:
ah ok 
If I use
[source,php]

it works.

Antora build was complaining about "[console,php]"